src/codecoverage/bot/tests/test_artifacts.py

```diff
@@ -27,6 +27,14 @@ def FAKE_ARTIFACTS_DIR(tmpdir):
     return tmpdir.strpath
 
 
+def test_generate_path(FAKE_ARTIFACTS_DIR):
+    a = ArtifactsHandler([], parent_dir=FAKE_ARTIFACTS_DIR)
+    artifact = {'name': 'code-coverage-jsvm.info'}
+    artifact2 = {'name': 'code-coverage-grcov.zip'}
+    assert a.parent_dir + '/linux_xpcshell-3_code-coverage-jsvm.info' == a.generate_path('linux', 'xpcshell-3', artifact)
+    assert a.parent_dir + '/windows_cppunit_code-coverage-grcov.zip' == a.generate_path('windows', 'cppunit', artifact2)
```

Comment:
Nit: using os.path.join would be preferable, instead of manually building the path